openhands/server/session/manager.py

```diff
@@ -13,16 +14,16 @@
 from openhands.storage.files import FileStore
 
 
+@dataclass
 class SessionManager:
-    _sessions: dict[str, Session] = {}
+    config: AppConfig
+    file_store: FileStore
     cleanup_interval: int = 300
     session_timeout: int = 600
+    _sessions: Dict[str, Session] = field(default_factory=dict)
+    _conversations: Dict[str, Conversation] = field(default_factory=dict)
```

Comment:
it appears that `Conversations` are 1:1 with `Session` by virtue of being key'ed on the same `sid`

Why keep it around in a different `dict` cache vs just as part of the `Session` object, as an optional
that way the garbage collection on closing out old sessions `_cleanup_sessions()` will take care of any potential leaks

Reply:
I update the cleanup method to clear conversations when sessions are cleared.

Reply:
I suppose a further refactor could include some construct where these are linked, but I need to be sure that these are in fact 1:1 first - At the moment a different method creates the conversation after the session.

Reply:
When running in HA mode, these are not 1:1.

The idea behind the Session/Conversations split (which is kind of a hack) is that a Session contains a long-lived websocket. The frontend is connected to a specific instance of OpenHands

But if we get a REST API request, in HA mode that request _might_ hit the same instance that's running the session, or it might not. So we create a Conversation object to attach to an existing runtime in order to serve that REST API call.

The Conversations should be _short_ lived--I don't think we should be keeping track of them at all